dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -287,6 +287,28 @@ def profile(self):
             profile.update(profiler.profile)
         return profile
 
+    def diff(self, other, options=None):
+        """
+        Finds the difference between 2 compilers and returns the report
+
+        :param other: profile compiler finding the difference with this one.
+        :type other: ColumnStatsProfileCompiler
+        :return: difference of the profiles
+        :rtype: dict
+        """
+        # Call super for compiler instance check
+        diff_profile = super().diff(other, options)
+
+        # Iterate through profiles
+        all_profiles = set(self._profiles.keys()) | set(other._profiles.keys())
+        if all_profiles:
```

Comment:
You shouldn't actually need this if, because if its just empty, then the for loop has nothing to loop through, and hence it won't happen anyway.